src/libraries/System.Data.Common/src/System/Data/DataSet.cs

```diff
@@ -1088,6 +1097,13 @@ public void Clear()
             }
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "Only parameterless constructors are used here but warning is about serialization constructor.")]
```

Comment:
See https://github.com/dotnet/runtime/pull/52046#discussion_r648054541

Reply:
Actually this one seems I can't avoid the suppression because Clone also calls into non-public ctors which means I need to add `DynamicallyAccessedMemberTypes.NonPublicConstructors` which is warning about serialization ctors (they are protected and not sure why it complains)

Reply:
Looking at the code, I believe what you have here is the best way we can make it. The only thing I'd say is maybe tweak the Justification a little

```suggestion
            Justification = "Only parameterless constructors are used here but since nonPublic=true, all non-public constructors are being preserved causing a warning for the serialization constructors. Those constructors won't be used here.")]
```

@vitek-karas - any thoughts?

Reply:
I agree - the fundamental problem is that DAM is not precise enough... but there's not much we can do about that here. The suppression should be correct - I like the justification proposed by @eerhardt .